codex-rs/core/src/codex.rs

```diff
@@ -1471,6 +1528,7 @@ fn maybe_run_with_user_profile(params: ExecParams, sess: &Session) -> ExecParams
 async fn handle_container_exec_with_params(
     params: ExecParams,
     sess: &Session,
+    turn_diff_tracker: &mut TurnDiffTracker,
```

Comment:
Hmm, what would happen if we wanted to support parallel tool calls at one point. This would be a problem, no?

Reply:
How do you think it would break?

Reply:
Because only one tool call could take ownership of TurnDiffTracker.

Reply:
I think we can solve this then. Maybe a Mutex or a channel or something, right?